src/utils/helpers.ts

```diff
@@ -0,0 +1,14 @@
+import { Person } from '../types';
+
+export const getFullDetailsOfPerson = (peoples: Person[]): Person[] => {
+  return peoples.map(person => {
+    const mother = peoples.find(p => p.name === person.motherName);
```

Comment:
The right decision, but now you have two searches in the middle of another search, which is algorithmically not optimal.
To improve the algorithm, you can create a separate map (`peopleMap`) and add all the people to it using their name as a key (`peopleMap.set(person.name, person)`), after that go through the `people` array and find the person's father and mother (`peopleMap.get(person.motherName)`, `peopleMap.get(person.fatherName)`).
This is a more complicated approach, but more correct on real projects.